R/getExperimentCrossCorrelation.R

```diff
@@ -0,0 +1,533 @@
+#' Calculate cross-correlation
+#' 
+#' @param x A
+#'   \code{\link[MultiAssayExperiment:MultiAssayExperiment-class]{MultiAssayExperiment}} or
+#'   \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#'   object.
+#' @param experiment1 experiment 1 (MAE)
+#' @param experiment2 experiment 2 (MAE and TreeSE)
+#' @param abund_values1 A single character value for selecting the
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assay}} to be
+#'   transformed.
+#' @param abund_values2 A single character value for selecting the
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assay}} to be
+#'   transformed.
+#' @param method association method ('pearson', or 'spearman'
+#' for continuous; categorical for discrete)
```

Comment:
Could we allow the user to provide their own function for calculating the pairwise associations as an input argument? This would provide a very flexible way for the users to experiment with different correlation techniques while taking advantage of these manipulation tools.

Reply:
I think that if we allow user to use own function, then only way to go is to just return "raw" associations as a table format.
 
Then user needs to do p-value adjustment and filtering, and additionally user needs to do matrix if it is wanted. We could provide examples though

Or then we could create a filtering function that gets information about columns as an input